src/DropDownMenu/DropDownMenu.jsx

```diff
@@ -343,57 +258,22 @@ const DropDownMenu = React.createClass({
 
     const styles = this.getStyles();
 
-    let value;
-    let selectedIndex = this._isControlled() ? null : this.state.selectedIndex;
-
-    if (menuItems && typeof selectedIndex === 'number') {
-      warning(menuItems[selectedIndex],
-        `SelectedIndex of ${selectedIndex} does not exist in menuItems.`);
-    }
-
-    if (valueMember && this._isControlled()) {
-      value = this.props.hasOwnProperty('value') ? this.props.value : valueLink.value;
-      if (menuItems && value !== null && value !== undefined) {
-        for (let i = 0; i < menuItems.length; i++) {
-          if (menuItems[i][valueMember] === value) {
-            selectedIndex = i;
-          }
-        }
-      }
-    }
-
     let displayValue = '';
-    if (menuItems) {
-      const selectedItem = menuItems[selectedIndex];
-      if (selectedItem) {
-        displayValue = selectedItem[labelMember || 'text'] || selectedItem[displayMember || 'text'];
+    React.Children.forEach(children, child => {
+      if (value === child.props.value) {
+        // This will need to be improved (in case primaryText is a node)
+        displayValue = child.props.label || child.props.primaryText;
       }
-    } else {
-      React.Children.forEach(children, child => {
-        if (value === child.props.value) {
-          // This will need to be improved (in case primaryText is a node)
-          displayValue = child.props.label || child.props.primaryText;
-        }
-      });
-    }
+    });
 
     let index = 0;
-    const menuItemElements = menuItems
-      ? menuItems.map((item, idx) => (
-        <MenuItem
-          key={idx}
-          primaryText={item[displayMember || 'text']}
-          value={item[valueMember]}
-          onTouchTap={this._onMenuItemTouchTap.bind(this, idx, item)}
-        />
-      ))
-      : React.Children.map(children, child => {
-        const clone = React.cloneElement(child, {
-          onTouchTap: this._onMenuItemTouchTap.bind(this, index, child.props.value),
-        }, child.props.children);
-        index += 1;
-        return clone;
-      });
+    const menuItemElements = React.Children.map(children, (child) => {
+      const clone = React.cloneElement(child, {
+        onTouchTap: this._onMenuItemTouchTap.bind(this, index, child.props.value),
+      }, child.props.children);
```

Comment:
That seems absurd to pass the children parameter. I'm gonna remove it, we wil save some CPU cycles: https://github.com/facebook/react/blob/v0.14.7/src/isomorphic/classic/element/ReactElement.js#L252.